### PR TITLE
[BE-05] Add comments

### PR DESCRIPTION
**Pull request by pannoewo**
https://trello.com/c/5G7moQjD

### CODE REVIEW

File: app/controllers/api/v1/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    class CommentsController < ApplicationController
+      before_action :set_recipe, only: %i[create]
+
+      def create
+        comment = Comment.new(comment_params)
+        comment.recipe = @recipe
+
+        if comment.save
+          render json: comment
+
```
Comment (plokijro) [inline app/controllers/api/v1/comments_controller.rb:12]:
```suggestion
          render json: comment
```

File: app/controllers/api/v1/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        else
+          render json: { errors: comment.errors.full_messages }, status: :unprocessable_entity
+
+        end
```
Comment (plokijro) [inline app/controllers/api/v1/comments_controller.rb]:
```suggestion
        if comment.save
          render json: comment
        else
          render json: { errors: comment.errors.full_messages }, status: :unprocessable_entity
        end
```

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  belongs_to :recipe
+  validates :body, presence: true, length: { minimum: 10 }
```
Comment (plokijro) [inline app/models/comment.rb]:
```suggestion
  belongs_to :recipe
  
  validates :body, presence: true, length: { minimum: 10 }
```

File: app/serializers/comment_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  attributes :id, :author, :body, :created_at
+  
+  ANNONYMOUS = "annonymous"
+  
+def author
+  return ANNONYMOUS if object.author.blank?
+  
+  object.author
+end 
+end
```
Comment (grabomarcin) [inline app/serializers/comment_serializer.rb]:
```suggestion
class CommentSerializer <ActiveModel::Serializer
  attributes :id, :author, :body, :created_at
  
  ANNONYMOUS = "annonymous"
  
  def author
    return ANNONYMOUS if object.author.blank?
  
    object.author
  end 
end

```

Please, fix syntax here

File: skrypt.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+
+IP_ADDR=`/sbin/ifconfig eth0 | grep 'inet addr' | cut -d: -f2 | awk '{print $1}'`
+
+echo "Twoj publiczny adres ip to: "
+curl http://169.254.169.254/latest/meta-data/public-ipv4 
+echo ""
+rails s -b $IP_ADDR -p 8082
```
Comment (grabomarcin) [inline skrypt.sh]:
```suggestion
rails s -b $IP_ADDR -p 8082

```
Add empty line at the end of file

File: app/serializers/recipe_serializer.rb [hunk 6]
```diff
@@ -1,3 +1,9 @@
 class RecipeSerializer < ActiveModel::Serializer
   attributes :id, :name, :content
+
+  has_many :comments
+
+  def comments
+    object.comments.order(created_at: :desc)
+  end
 end
```
Comment (plokijro) [inline app/serializers/recipe_serializer.rb]: Comments should be returned only in show recipe response

File: db/migrate/20220713071422_create_comments.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[6.1]
+  def change
+    create_table :comments do |t|
+      t.string :author #dodajemu autora , jesli go nie ma "unnown"
```
Comment (plokijro) [inline db/migrate/20220713071422_create_comments.rb]: 👀

File: spec/controllers/api/v1/comments_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  describe '[POST] #create' do
+    let(:recipe) { create :recipe }
+    let(:post_request) { post :create, params: recipe_params }
+
+    context 'with correct params' do
+      let(:recipe_params) do
+        {
+          comment: {
+            author: 'Adam Mickiewicz',
+            body: 'nininniimijijjijij'
```
Comment (grabomarcin) [inline spec/controllers/api/v1/comments_controller_spec.rb]: You can write here some real text 😎

File: spec/factories/comments.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :comment do
+    author {Faker::Name.name}
+    body {Faker::ChuckNorris.fact }
+    recipe 
+end
+end
```
Comment (grabomarcin) [inline spec/factories/comments.rb]:
```suggestion
FactoryBot.define do
  factory :comment do
    author {Faker::Name.name}
    body {Faker::ChuckNorris.fact }
    recipe 
  end
end

```
Fix syntax

File: spec/models/comment_spec.rb [hunk 10]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  describe 'asociate' do
```
Comment (grabomarcin) [inline spec/models/comment_spec.rb]:
```suggestion
  describe 'associations' do
```

File: spec/models/recipe_spec.rb [hunk 11]
```diff
@@ -1,6 +1,10 @@
 require 'rails_helper'
 
 RSpec.describe Recipe, type: :model do
+  describe 'associantions' do
```
Comment (grabomarcin) [inline spec/models/recipe_spec.rb]:
```suggestion
  describe 'associations' do
```

File: spec/serializers/comment_serializer_spec.rb [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    
+end
+
+   context "when commnet without author" do
+     let(:comment) {build :comment, author: nil}
+     
+     it { is_expected.to include(author: 'annonymous')}
+   end   
+ end 
+ end
```
Comment (grabomarcin) [inline spec/serializers/comment_serializer_spec.rb]: Run rubocop on this file to fix syntax

File: app/serializers/comment_serializer.rb [hunk 13]
```diff
@@ -0,0 +1,10 @@
+class CommentSerializer < ActiveModel::Serializer
+  ANNONYMOUS = 'annonymous'.freeze
+  attributes :id, :author, :body, :created_at
```
Comment (grabomarcin) [inline app/serializers/comment_serializer.rb]:
```suggestion
class CommentSerializer < ActiveModel::Serializer
  ANNONYMOUS = 'annonymous'.freeze
  
  attributes :id, :author, :body, :created_at
```

File: app/serializers/recipe_serializer.rb [hunk 14]
```diff
@@ -1,3 +1,9 @@
 class RecipeSerializer < ActiveModel::Serializer
   attributes :id, :name, :content
+
+  has_many :comments
+
+  def comments
+    object.comments.order(created_at: :desc)
+  end
```
Comment (grabomarcin) [inline app/serializers/recipe_serializer.rb]: You should remove that here

File: app/serializers/comment_serializer.rb [hunk 15]
```diff
@@ -1,5 +1,6 @@
 class CommentSerializer < ActiveModel::Serializer
-  ANNONYMOUS = 'annonymous'.freeze
+  ANNONYMOUS = 'annonymous'
```
Comment (grabomarcin) [inline app/serializers/comment_serializer.rb]:
```suggestion
  ANNONYMOUS = 'annonymous'.freeze
```

File: spec/factories/comments.rb [hunk 16]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :comment do
+    author { Faker::Name.name }
+    body { Faker::ChuckNorris.fact }
+    recipe
```
Comment (plokijro) [inline spec/factories/comments.rb]:
```suggestion
    recipe
    author { Faker::Name.name }
    body { Faker::ChuckNorris.fact }
```

Comment (plokijro) [reply spec/factories/comments.rb]: Associations should be before attributes

File: db/migrate/20220713071422_create_comments.rb (hunk 7 above)
Comment (plokijro) [reply db/migrate/20220713071422_create_comments.rb:4]: Remove comment please

File: app/models/comment.rb [hunk 17]
```diff
@@ -0,0 +1,5 @@
+class Comment < ApplicationRecord
+  belongs_to :recipe
+
+  validates :body, presence: true, length: { minimum: 10 }
+end
```
Comment (plokijro) [inline app/models/comment.rb]: I think we should add length validation for `author`

File: db/migrate/20220713071422_create_comments.rb (hunk 7 above)
Comment (plokijro) [inline db/migrate/20220713071422_create_comments.rb]: Please remove comment

**Review by grabomarcin — APPROVED** [review]

**Pull request merged**